modules/secrets/main.tf

```diff
@@ -0,0 +1,77 @@
+##############################################################################
+# Secret Group
+##############################################################################
+
+locals {
+  secret_groups = flatten([
+    for secret_group in var.secrets :
+    secret_group.existing_secret_group || secret_group.secret_group_name == null ? [] : [{
```

Comment:
secret_group_name is required, can it ever be null?

Reply:
when we set `secret_group_name ` to null, the secret will be created in the `default` secret group.